### PR TITLE
[FIXED] Possible panic when server accepts TLS leafnode connection

### PR DESCRIPTION
**Pull request by kozlovic**
Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: server/leafnode.go [hunk 1]
```diff
@@ -649,6 +649,14 @@ func (s *Server) createLeafNode(conn net.Conn, remote *leafNodeCfg) *client {
 			return nil
 		}
 
+		// Not sure that can happen, but in case the connection was marked
+		// as closed during In case connection was marked as closed during parse...
```
Comment (derekcollison) [inline server/leafnode.go:653]: In -> in

**Review by derekcollison — COMMENTED** [review]
Does this solve for the panic though and the stack trace we saw in Slack?

File: server/leafnode.go (hunk 1 above)
Comment (kozlovic) [reply server/leafnode.go:653]: whole comment is wrong :-) will fix that.

**Review by philpennock — APPROVED** [review]

**Comment by kozlovic** [comment]
@derekcollison The panic was for the non solicited. When calling `sendProtoNow()` this could have marked the connection as closed right there if there was a TCP write error.

**Comment by kozlovic** [comment]
Pushing the change for the wrong comment..

File: server/leafnode.go [hunk 2]
```diff
@@ -649,6 +649,14 @@ func (s *Server) createLeafNode(conn net.Conn, remote *leafNodeCfg) *client {
 			return nil
 		}
 
+		// Not sure that can happen, but in case the connection was marked
+		// as closed during the call to parse...
+		if c.isClosed() {
```
Comment (derekcollison) [inline server/leafnode.go:654]: Let's make this a small anonymous function inside this function to clean up?

`ifClosedReturn := func() `

**Review by derekcollison — APPROVED** [review]
LGTM, maybe make that a small internal function inside the create function.

File: server/leafnode.go (hunk 2 above)
Comment (kozlovic) [reply server/leafnode.go:654]: I would still need to have something like:
```
if closedReturn() {
   return nil
}
```

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**